MarvelCharsApp/MarvelCharsAppTests/HomeViewModelTest.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  HomeViewModelTest.swift
+//  MarvelCharsAppTests
+//
+//  Created by admin on 11/30/21.
+//
+
+import XCTest
+
+class HomeViewModelTest: XCTestCase {
+
+    override func setUpWithError() throws {
+        // Put setup code here. This method is called before the invocation of each test method in the class.
+    }
+
+    override func tearDownWithError() throws {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+    }
+
+  func testGetCategoryWithName() throws {
+    let viewModel = HomeViewModel()
+    viewModel.categories.append(CategoryModel(category: "Heroes"))
+    viewModel.categories.append(CategoryModel(category: "AntiHeroes"))
+    viewModel.categories.append(CategoryModel(category: "Humans"))
+    let category = viewModel.getCategoryWith(name: "AntiHeroes")
+    XCTAssertEqual(category?.category, "AntiHeroes")
+  }
+  
+  func testReturnNilWhenGetCategoryWithWrongName() throws {
+    let viewModel = HomeViewModel()
+    viewModel.categories.append(CategoryModel(category: "Heroes"))
```

Comment:
The data should not be mocked this way on the viewModel. You need to create a mock Manager and pass it as a value to the model. The viewmodel should call fetchData here and call this manager to obtain the fake data.
The declaration of that fake manager can be in this class.